packages/fast-form-generator-react/src/form/form.utilities.ts

```diff
@@ -218,6 +221,160 @@ export function getComponentTracker(
+    const dataLocationSegments: string[] = normalizedDataLocation.split(".");
+    const schemaLocationSegments: string[] = getSchemaLocationSegmentsFromDataLocationSegments(dataLocationSegments, schema, data);
+
+    return normalizeSchemaLocation(schemaLocationSegments.join("."));
+}
+
+/**
+ * Get an array of schema location strings from an array of data location strings
+ */
+export function getSchemaLocationSegmentsFromDataLocationSegments(dataLocationSegments: string[], schema: any, data: any): string[] {
+    let schemaLocationSegments: string[] = getSchemaOneOfAnyOfLocationSegments(schema, data);
+
+    for (let i: number = 0; i < dataLocationSegments.length; i++) {
+        const partialData: any = getPartialData(dataLocationSegments.slice(0, i).join("."), data);
+        const partialSchema: any = getPartialData(normalizeSchemaLocation(schemaLocationSegments.join(".")), schema);
+
+        schemaLocationSegments = schemaLocationSegments.concat(
+            getSchemaOneOfAnyOfLocationSegments(
+                partialSchema,
+                partialData
+            )
+        );
+
+        schemaLocationSegments = schemaLocationSegments.concat(
+            getSchemaLocationSegmentsFromDataLocationSegment(
+                dataLocationSegments[i],
+                dataLocationSegments.length === i + 1,
+                partialData
+            )
+        );
+    }
+
+    return schemaLocationSegments;
+}
+
+/**
+ * Gets data from a data and location
+ */
+export function getPartialData(location: string, data: any): any {
+    return location === "" ? data : get(data, location);
+}
+
+/**
+ * Gets an array of oneOf/anyOf with a valid index from a schema and data
+ */
+export function getSchemaOneOfAnyOfLocationSegments(schema: any, data: any): string[] {
+    const schemaLocationSegments: string[] = [];
+
+    if (!!schema.anyOf) {
+        schemaLocationSegments.push(`anyOf.${getValidAnyOfOneOfIndex("anyOf", data, schema)}`);
+    }
+
+    if (!!schema.oneOf) {
+        schemaLocationSegments.push(`oneOf.${getValidAnyOfOneOfIndex("oneOf", data, schema)}`);
+    }
+
+    return schemaLocationSegments;
+}
+
+/**
+ * Get an array of schema location strings from a single data location item
+ */
+export function getSchemaLocationSegmentsFromDataLocationSegment(
+    dataLocation: string,
+    lastDataLocationSegment: boolean,
+    data: any
+): string[] {
+    const schemaLocationSegments: string[] = [];
+
+    if (typeof data === "object" && data !== null && !Array.isArray(data)) {
+        schemaLocationSegments.push("properties");
+    }
+
+    schemaLocationSegments.push(dataLocation);
+
+    // In the case that this is an array and this is not the the last item in the data location string,
+    // add the JSON schema "items" keyword
+    if ((Array.isArray(data) || isDataLocationArrayItem(dataLocation)) && !lastDataLocationSegment) {
+        schemaLocationSegments.push("items");
```

Comment:
assuming because it's a "keyword" per your comment that "items" doesn't need to be localized.